scripts/safari.js

```diff
@@ -1156,6 +1113,58 @@ function Safari() {
             return result.slice(-upper, -lower);
         }
     }
+    function removeDuplicates(arr) {
+        var result = {};
+        for (var x in arr) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.